src/react-components/src/components/03-organisms/Headers/HeaderWithSearch.tsx

```diff
@@ -0,0 +1,78 @@
+// OH-31 Header with Search
+import * as React from "react";
+import SearchBar from "../../02-molecules/SearchBar/SearchBar";
+import UnderlineLink from "../../01-atoms/Links/UnderlineLink";
+import bem from "../../../utils/bem";
+import BodyText from "../../01-atoms/Text/Text/BodyText";
+import SectionTitle from "../../02-molecules/Headings/SectionTitle";
+import Heading from "../../01-atoms/Text/Headings/Heading";
+import RNSectionTitle from "../../01-atoms/Text/Headings/RNSectionTitle";
+
+
+export interface WithSearchProps {
+  modifiers?: [];
+  blockName?: string;
+  searchBarId: string;
+  searchButtonId: string;
+  searchBarAriaLabel: string;
+  dropdownId: string;
+  headingContent: JSX.Element;
+  headingId: string;
+  headingBaseClass?: string;
+  headingUrl?: string;
+  textFieldAriaLabel: string;
+  hasError?: boolean;
+  // To make into jsx.element (eg: if error has link)
+  // must first handle the placehodlerText=errorMessage in ResearchNow's Search Header
+  errorMessage?: string;
+  searchDropdownOptions: string[];
+  advancedSearchElem: JSX.Element;
+  selectChangeHandler: (event: React.FormEvent) => void;
+  selectBlurHandler: (event: React.FormEvent) => void;
+  searchSubmitHandler: (event: React.MouseEvent) => void;
+  textChangeHandler?: (event: React.FormEvent) => void;
+}
+
+export default function HeaderWithSearch(props: React.PropsWithChildren<WithSearchProps>) {
+
+  const { blockName, searchBarId, searchBarAriaLabel,
+    searchButtonId, hasError, errorMessage,
+    dropdownId, searchDropdownOptions,
+    headingContent, headingId, headingUrl, headingBaseClass,
+    textFieldAriaLabel, advancedSearchElem,
+    selectChangeHandler, selectBlurHandler, searchSubmitHandler, textChangeHandler } = props;
+  const base_class = "search-header";
+
+  React.cloneElement(advancedSearchElem, {modifiers: ["dark-background"]});
+
+  return (
+    <div className={bem(base_class)}>
+      <div className={bem("content", [], base_class)}>
+        <RNSectionTitle
+          blockName={base_class}
+          id={headingId}
+          url={headingUrl} />
```

Comment:
Because you use a non-generic component, I think `RNHeaderWithSearch` is a more appropriate name for this component. I imagine this will work great for RN, but for other projects there may be confusion if this `HeaderWithSearch` component is pulled in and it renders "ResearchNOW". I do get this should be out rather quickly so if this name will be temporary, that's fine, but it should be reconsidered.

Reply:
yeah that's a better approach to thinking about this work.

Reply:
I agree but we should be matching Metronome, as its intended to be our source of truth. I'll crosspost this to Slack to discuss.